### ISSUE TITLE
Cannot bind AccessToken parameter error

### ISSUE THREAD

**Issue by jarrydstan** [Sekers/Useful-Scripts#5]
**Describe the bug**
When running the script on Powershell 7 with a ClientSecret, it's getting an error on line 286 with the error code 

`Cannot bind parameter 'AccessToken'. Cannot convert the value of type "System.String" to type "System.Security.SecureString".`

**To Reproduce**

- use ClientSecret with the Encrypted string from the provided script
- Run the sync script
- error

**Expected behavior**
Accesstoken clears successfullly

**Desktop (please complete the following information):**
 - OS: Windows Server 2022
 - PowerShell Edition PowerShell 7
 - Version 7.4.6

**Comment by Sekers** [comment]
Which script?
@jarrydstan

**Comment by jarrydstan** [comment]
@Sekers Sorry mate, it's the M365 Teams Membership Sync, I managed to get around it by using ($AccessToken |ConvertTo-SecureString -AsPlainText -Force) but obviously that's not too secure! 😄

**Comment by Sekers** [comment]
I highly recommend using certificate auth rather than putting a secret in the config, but for testing it should work. Make sure that you encrypt the secret and put the encrypted string version in the config, not the plain text.

Check out the README for the script under "MgApp_EncryptedSecret" for more information. You can use the example script linked there to create an encrypted version. From there, the script must be run from the same account on the same machine as where you encrypted the string, otherwise it won't decrypt.

**Comment by Sekers** [comment]
Closing as it's not a bug. The secret placed into the configuration file needs to be encrypted and the context the script runs in needs to have the ability to decrypt the encrypted string.

**Comment by jarrydstan** [comment]
@Sekers sorry didnt have a chance to respond, the clientsecret was encrypted using your other script using my user account, and regardless of whether i run it as myself or admin, it still fails, only by converting to plaintext does it work.

Thanks

**Comment by Sekers** [comment]
@jarrydstan Got it. I looked into this more and the last time I tested using Client Secret was with PowerShell Graph 1.x. The current code still works with that version of Graph SDK, but It looks like something has changed with the 2.x version.

**Comment by Sekers** [comment]
Connect-MgGraph -AccessToken $AccessToken

`Connect-MgGraph : Cannot bind parameter 'AccessToken'. Cannot convert the "string here" value of type "System.String" to type
"System.Security.SecureString".`

**Comment by jarrydstan** [comment]
@Sekers appreciate your quick responses! Let me jnow if you need me to test anything

**Comment by Sekers** [comment]
@jarrydstan Resolved issue with commit cda0fe4197ce7e7540e29abf4e48d7493e794bbc

The Microsoft Graph PowerShell SDK v2 has added the 'ClientSecretCredential' parameter so connecting is now even easier.

```
$ClientSecretCredential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList $MgClientID, $($Config.General.MgApp_EncryptedSecret | ConvertTo-SecureString)
                        $null = Connect-MgGraph -TenantId $MgTenantID -ClientSecretCredential $ClientSecretCredential
```
